src/librbd/api/Mirror.cc

```diff
@@ -437,6 +437,10 @@ int Mirror<I>::image_enable(I *ictx, mirror_image_mode_t mode,
   if (mirror_mode != cls::rbd::MIRROR_MODE_IMAGE) {
     lderr(cct) << "cannot enable mirroring in the current pool mirroring mode"
                << dendl;
+    if (!ictx->md_ctx.get_namespace().empty()) {
+      lderr(cct) << "check if mirroring is enabled on namespace: "
+                 << ictx->md_ctx.get_namespace() << dendl;
```

Comment:
We don't generally report multiple errors like this.  Further, the existing `cannot enable mirroring in the current pool mirroring mode` error message remains confusing because it piles `MIRROR_MODE_DISABLED` and `MIRROR_MODE_POOL` cases together.

I think it would take four different error messages to improve this:
- "cannot enable mirroring: mirroring is not enabled on a pool"
- "cannot enable mirroring: mirroring is not enabled on a namespace"
- "cannot enable mirroring: pool is not in image mirror mode"
- "cannot enable mirroring: namespace is not in image mirror mode"

Otherwise it's probably not worth the churn.